plugins/scanners/scancode/src/main/kotlin/ScanCodeResultParser.kt

```diff
@@ -34,16 +42,45 @@ import org.ossreviewtoolkit.model.TextLocation
 import org.ossreviewtoolkit.model.createAndLogIssue
 import org.ossreviewtoolkit.model.mapLicense
 import org.ossreviewtoolkit.model.utils.associateLicensesWithExceptions
-import org.ossreviewtoolkit.utils.common.textValueOrEmpty
-import org.ossreviewtoolkit.utils.spdx.SpdxConstants.LICENSE_REF_PREFIX
+import org.ossreviewtoolkit.utils.spdx.SpdxConstants
 import org.ossreviewtoolkit.utils.spdx.toSpdxId
 
 import org.semver4j.Semver
 
 const val MAX_SUPPORTED_OUTPUT_FORMAT_MAJOR_VERSION = 2
 
-internal val SCANCODE_TIMESTAMP_FORMATTER: DateTimeFormatter =
-    DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HHmmss.n").withZone(ZoneId.of("UTC"))
+private val LICENSE_REF_PREFIX_SCAN_CODE = "${SpdxConstants.LICENSE_REF_PREFIX}${ScanCode.SCANNER_NAME.lowercase()}-"
+private val TIMESTAMP_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HHmmss.n").withZone(ZoneId.of("UTC"))
+
+fun parseResult(result: File) = parseResult(result.readText())
+
+fun parseResult(result: String) = parseResult(Json.parseToJsonElement(result))
+
+fun parseResult(result: JsonElement): ScanCodeResult {
+    // As even the structure of the header itself may change with the output format version, first operate on raw JSON
+    // elements to get the version, and then parse the JSON elements into the appropriate data classes.
+    val header = result.jsonObject.getValue("headers").jsonArray.single().jsonObject
+    val outputFormatVersion = header["output_format_version"]?.let { Semver(it.jsonPrimitive.content) }
+
+    // Select the correct set of (de-)serializers bundled in a module for parsing the respective format version.
+    val module = when (outputFormatVersion?.major) {
+        null, 1 -> SerializersModule {
+            polymorphicDefaultDeserializer(CopyrightEntry::class) { CopyrightEntry.Version1.serializer() }
+        }
+
+        else -> SerializersModule {
+            polymorphicDefaultDeserializer(CopyrightEntry::class) { CopyrightEntry.Version2.serializer() }
+        }
+    }
+
+    val json = Json {
+        ignoreUnknownKeys = true
+        namingStrategy = JsonNamingStrategy.SnakeCase
+        serializersModule = module
+    }
```

Comment:
This could be a constant to reuse it when parsing multiple results.

Reply:
You mean the whole `Json` instance? I believe that's not possible as the modules are set dynamically depending on the output format version contained in each file.